gtests/net/mptcp/add_addr/add_addr_port_client.pkt

```diff
@@ -0,0 +1,25 @@
+// connection initiated by the kernel
+--tolerance_usecs=100000
+`../common/defaults.sh`
+
++0     `../common/client.sh`
+
++0.0   socket(..., SOCK_STREAM, IPPROTO_MPTCP) = 3
++0.0   setsockopt(3, SOL_SOCKET, SO_REUSEADDR, [1], 4) = 0
++0.0   fcntl(3, F_GETFL) = 0x2 (flags O_RDWR)
++0.0   fcntl(3, F_SETFL, O_RDWR|O_NONBLOCK) = 0
+
++0.0   connect(3, ..., ...) = -1 EINPROGRESS (Operation now in progress)
++0.0      >                             S   0:0(0)                    <mss 1460, sackOK, TS val 100 ecr 0,   nop, wscale 8, mpcapable v1 flags[flag_h] nokey>
++0.0      <                             S.  0:0(0)  ack 1  win 65535  <mss 1460, sackOK, TS val 700 ecr 100, nop, wscale 8, mpcapable v1 flags[flag_h] key[skey=2]>
++0.0      >                              .  1:1(0)  ack 1             <nop, nop,         TS val 100 ecr 700,                mpcapable v1 flags[flag_h] key[ckey, skey]>
+
++0.200  getsockopt(3, SOL_SOCKET, SO_ERROR, [0], [4]) = 0
++0.205  fcntl(3, F_SETFL, O_RDWR) = 0   // set back to blocking
+
+// ADD_ADDRESS with port
++0.0      <                              .  1:1(0)  ack 1 win 256                                                          <add_address address_id=12 addr[saddr] port=4321 hmac=auto, dss dack4=1 dll=0 nocs, nop, nop>
++0.0      >                              .  1:1(0)  ack 1             <nop, nop,          TS val 100 ecr 700,               add_address address_id=12 addr[saddr] port=4321 addr_echo, nop, nop>
+
+// TODO validate destination port in the outbound mp_join syn
+//+1.0 >    > addr[saddr]                  S  0:0(0)                    <mss 1460, sackOK, TS val 100 ecr 0, nop, wscale 8,   mp_join_syn address_id=0 token=sha256_32(skey)>
```

Comment:
I think that packetdrill already has the possibility to specify ports and address. I didn't understand yet how to do that with environment variables.

Reply:
but can we do it only for one subflow?

e.g. here we would have the initial subflow using the default port 8080 (we don't care about this port) but we want to check that the second subflow will be done over the port 4321. (maybe we don't need to go that far for the moment and we just want to check the ADD_ADDR is properly sent because I think it might be difficult to support subflows using different ports)

Reply:
> but can we do it only for one subflow?
> 
> e.g. here we would have the initial subflow using the default port 8080 (we don't care about this port) but we want to check that the second subflow will be done over the port 4321. (maybe we don't need to go that far for the moment and we just want to check the ADD_ADDR is properly sent because I think it might be difficult to support subflows using different ports)

I think (but I need to verify it)  that the current code is already usable for that. From what I saw in the past, when a live packet is seen with a 5-uple that does not match the current socket, a new one is allocated and marked as current.